File: app/src/main/java/io/github/sithengineer/motoqueiro/data/RideDataSource.java
package io.github.sithengineer.motoqueiro.data;

import io.github.sithengineer.motoqueiro.data.model.GpsPoint;
import io.github.sithengineer.motoqueiro.data.model.HeartRatePoint;
import io.github.sithengineer.motoqueiro.data.model.Ride;
import io.github.sithengineer.motoqueiro.data.model.TriDimenPoint;
import java.util.List;

public interface RideDataSource {
  void saveGpsData(String rideId, List<GpsPoint> points);

  long saveGpsData(String rideId, GpsPoint coords);

  List<GpsPoint> getGpsData(String rideId);

  void saveAccelerometerData(String rideId, List<TriDimenPoint> points);

  long saveAccelerometerData(String rideId, TriDimenPoint point);

  List<TriDimenPoint> getAccelerometerData(String rideId);

  void saveGyroscopeData(String rideId, List<TriDimenPoint> points);

  long saveGyroscopeData(String rideId, TriDimenPoint point);

  void saveGravityData(String rideId, List<TriDimenPoint> points);

  long saveGravityData(String rideId, TriDimenPoint point);

  List<TriDimenPoint> getGravityData(String rideId);

  void saveHeartRateData(String rideId, List<HeartRatePoint> points);

  long saveHeartRateData(String rideId, HeartRatePoint point);

  List<HeartRatePoint> getHeartRateData(String rideId);

  long saveRide(Ride ride);

  List<TriDimenPoint> getGyroscopeData(String rideId);

  Ride getRide(String rideId);

  boolean markCompleted(String rideId);

  List<Ride> getCompletedRides();

  boolean markSynced(String rideId);
}

File: app/src/main/java/io/github/sithengineer/motoqueiro/data/RideDataSource.kt
package io.github.sithengineer.motoqueiro.data

import io.github.sithengineer.motoqueiro.data.local.entity.AccelerometerPoint
import io.github.sithengineer.motoqueiro.data.local.entity.GpsPoint
import io.github.sithengineer.motoqueiro.data.local.entity.GravityPoint
import io.github.sithengineer.motoqueiro.data.local.entity.GyroscopePoint
import io.github.sithengineer.motoqueiro.data.local.entity.HeartRatePoint
import io.github.sithengineer.motoqueiro.data.local.entity.Ride
import io.reactivex.Completable
import io.reactivex.Single

interface RideDataSource {

  fun saveGpsData(points: List<GpsPoint>): Completable

  fun saveGpsData(coordinates: GpsPoint): Completable

  fun getGpsData(rideId: Long): Single<List<GpsPoint>>

  fun saveAccelerometerData(points: List<AccelerometerPoint>): Completable

  fun saveAccelerometerData(point: AccelerometerPoint): Completable

  fun getAccelerometerData(rideId: Long): Single<List<AccelerometerPoint>>

  fun saveGyroscopeData(points: List<GyroscopePoint>): Completable

  fun saveGyroscopeData(point: GyroscopePoint): Completable

  fun getGyroscopeData(rideId: Long): Single<List<GyroscopePoint>>

  fun saveGravityData(points: List<GravityPoint>): Completable

  fun saveGravityData(point: GravityPoint): Completable

  fun getGravityData(rideId: Long): Single<List<GravityPoint>>

  fun saveHeartRateData(points: List<HeartRatePoint>): Completable

  fun saveHeartRateData(point: HeartRatePoint): Completable

  fun getHeartRateData(rideId: Long): Single<List<HeartRatePoint>>

  fun saveRide(ride: Ride): Completable

  fun getRide(rideId: Long): Single<Ride>

  fun getCompletedRides(): Single<List<Ride>>

  fun markCompleted(rideId: Long): Completable

  fun markSynced(rideId: Long): Completable
}
